File: python/src/epostak/resources/account.py
"""Account resource -- account info and usage.

Provides :class:`AccountResource` for retrieving the authenticated account's
firm details, subscription plan, and current billing-period usage.

For key introspection, OAuth token minting, and key rotation see
``client.auth.*`` (:class:`~epostak.resources.auth.AuthResource`).
"""

from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from epostak.types import Account

from epostak.resources.documents import _BaseResource


class AccountResource(_BaseResource):
    """Account and firm information."""

    def get(self) -> Account:
        """Get account info including firm details, plan, and usage.

        Returns:
            Dict with ``firm`` (name, ICO, Peppol status), ``plan`` (name,
            status), ``usage`` (outbound/inbound counts for the current
            billing period), and ``limits`` (per-plan caps).

        Example::

            account = client.account.get()
            print(account["firm"]["name"], account["plan"]["name"])
            print(f"Sent: {account['usage']['outbound']}")
        """
        return self._request("GET", "/account")

    def license_info(self) -> dict:
        """Get per-firm plan and current-period license usage."""
        return self._request("GET", "/licenses/info")

File: ruby/lib/epostak/resources/account.rb
# frozen_string_literal: true

module EPostak
  module Resources
    # Resource for retrieving account information — firm details, subscription
    # plan, and document usage for the current billing period.
    #
    # For key introspection, OAuth token minting, and key rotation see
    # `client.auth.*`. The v1 helpers `client.account.status` and
    # `client.account.rotate_secret` were removed in 2.0.0.
    #
    # @example
    #   account = client.account.get
    #   puts "Plan: #{account['plan']['name']} (#{account['plan']['status']})"
    #   puts "Usage: #{account['usage']['outbound']} sent, #{account['usage']['inbound']} received"
    class Account
      # @param http [EPostak::HttpClient] Internal HTTP client
      def initialize(http)
        @http = http
      end

      # Get account information for the authenticated API key.
      # Returns the associated firm, current subscription plan, and usage counters.
      #
      # @return [Hash] Account details with "firm", "plan", and "usage" sections
      #
      # @example
      #   account = client.account.get
      #   if account["plan"]["status"] == "expired"
      #     puts "Subscription expired -- renew to continue sending documents"
      #   end
      def get
        @http.request(:get, "/account")
      end
    end
  end
end
